### ISSUE TITLE
Rightwards keyboard navigation doesn't work if selected file is at right edge of view.

### ISSUE THREAD

**Issue by uncertainquark** [elementary/files#391]
If I am keyboard navigating files in a folder like this:
![screenshot from 2018-03-24 16-31-25](https://user-images.githubusercontent.com/22585732/37863242-0c102600-2f81-11e8-9f05-93c17ecf4fa4.png)

Then the "Down" keyboard button doesn't work if there is no file below the selected file i.e. it doesn't select the file below and refuses to navigate from its existing position.

**Comment by jeremypw** [comment]
You should be able to press right arrow to select the next file - but these seems not to be working - I think this is a regression. I would not expect the down arrow to work under these circumstances.

Thanks for taking the time to file some issues :-).

**Comment by jeremypw** [comment]
Seems to be effectively a duplicate of #362

**Comment by uncertainquark** [comment]
>  I would not expect the down arrow to work under these circumstances.

All other file managers that I have tried (Windows, Nautilus, Dolphin) do select the file in the next row. I think it is an expected behavior because one is trying to scroll a list and as such the navigation should take the user till the end of the list and not stop abruptly.

**Comment by jeremypw** [comment]
OK, I'll see what the UX team think.

**Comment by jeremypw** [comment]
Note: Thunar behaves as Files does when down arrow pressed under these circumstances

**Comment by uncertainquark** [comment]
> Note: Thunar behaves as Files does when down arrow pressed under these circumstances. It is the native behavior of the Gtk.IconView widget used.

Sure. Although I think since the primary audience of elementary comes from Windows, Mac and I guess Ubuntu, it means the UX behavior there needs to be kept in mind. Thunar is sparsely used.
